Library/Homebrew/cmd/--selfdestruct.rb

```diff
@@ -0,0 +1,93 @@
+# Original Inspiration: Stephen Benner https://github.com/SteveBenner: https://gist.github.com/SteveBenner/11254428
+# Partially Rewritten by Dominyk Tiller https://github.com/DomT4 for Homebrew's internal use.
+# THIS WILL COMPLETELY REMOVE HOMEBREW FROM YOUR SYSTEM
+
+require "fileutils"
+require "open3"
+require "rubygems"
+
+begin
+  require "trollop"
+rescue LoadError
+  onoe "You must run `sudo /System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/gem install trollop`"
+  raise
+end
+
+module Homebrew
+  def __selfdestruct
+
+@options = Trollop::options do
+  opt :quiet, "Run script with minimal user-visible output"
+  opt :verbose, "Run script with maximum user-visible output"
+  opt :dryrun, "Don't delete anything. Just show what would happen"
+end
+
+brew_location = HOMEBREW_PREFIX
+
+brewlocal = %W{.git Cellar Library/brew.rb Library/Homebrew Library/Aliases Library/Formula Library/Contributions Library/ENV Library/LinkedKegs}
+$files = []
+
+# Found Homebrew installation
+if brew_location
+  unless @options[:quiet]
+    puts "Homebrew found at: #{brew_location}"
+    begin # record kegs and taps for later output
+      brewed = `brew list`
+      tapped = `brew tap`
+    rescue StandardError
+    end
+  end
+  # Collect files indexed by git
+  begin
+    Dir.chdir(brew_location) do
+      # Update file list (use popen3 so we can suppress git error output)
+      Open3.popen3('git checkout master') { |stdin, stdout, stderr| stderr.close }
+      $files += `git ls-files`.split.map {|file| File.expand_path file }
+    end
+  rescue StandardError => e
+    puts e
+  end
+end
+
+# Collect files
+$files += brewlocal.select { |file| File.exist? file }.map {|file| File.expand_path file }
+cache = "#{HOMEBREW_CACHE}"
+logs = "#{HOMEBREW_LOGS}"
+
+abort 'Failed to locate any homebrew files!' if $files.empty?
+
+rm =
+  if @options[:dryrun]
```

Comment:
There's an extremely minimal amount of interactivity in the install script as is. It's pretty much the bare minimum along the lines of ` Hey we're gonna install Homebrew right now, you cool with that? `, which we could do here as well, but I'm not sure what use interactivity would have in an uninstall script beyond that. Happy to take other views though, Don't want to flood the entire thread with our discussion :smile_cat: